tools/nfdc/nfdc_status.go

```diff
@@ -0,0 +1,55 @@
+package nfdc
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	enc "github.com/named-data/ndnd/std/encoding"
+	mgmt "github.com/named-data/ndnd/std/ndn/mgmt_2022"
+)
+
+func (n *Nfdc) ExecStatusGeneral(args []string) {
+	suffix := enc.Name{
+		enc.NewStringComponent(enc.TypeGenericNameComponent, "status"),
+		enc.NewStringComponent(enc.TypeGenericNameComponent, "general"),
+	}
+
+	data, err := n.fetchStatusDataset(suffix)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error fetching status dataset: %+v\n", err)
+		return
+	}
+
+	status, err := mgmt.ParseGeneralStatus(enc.NewWireReader(data), true)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error parsing general status: %+v\n", err)
+		return
+	}
+
+	fmt.Println("General NFD status:")
+	n.statusPadding = 24
+	n.printStatusLine("version", status.NfdVersion)
+	n.printStatusLine("startTime", time.UnixMilli(int64(status.StartTimestamp)))
```

Comment:
You can use TimeField:
https://github.com/named-data/ndnd/blob/e6a187018765a263efb5222e0f802056506415f5/std/encoding/tests/gen_basic/definition.go#L13-L14

Reply:
Hmm but this is a millisecond epoch and we support time?

Reply:
Time field does a conversion. When encoding, it becomes millisecond epoch
https://github.com/named-data/ndnd/blob/e6a187018765a263efb5222e0f802056506415f5/std/encoding/codegen/fields_time.go#L45

And it converts the number back to `Duration` after decoding

https://github.com/named-data/ndnd/blob/e6a187018765a263efb5222e0f802056506415f5/std/encoding/codegen/fields_time.go#L63

Reply:
Awesome! Changed.